design-documents/graph-ql/coverage/b2b/purchase-order.graphqls

```diff
@@ -0,0 +1,303 @@
+    rejectPurchaseOrder(input: RejectPurchaseOrderInput!): RejectPurchaseOrderOutput @doc(description: "Reject purchase order")
+    createPurchaseOrderApprovalRule(input: CreatePurchaseOrderApprovalRuleInput!): CreatePurchaseOrderApprovalRuleOutput @doc(description: "Create purchase order approval rule")
+    updatePurchaseOrderApprovalRule(input: UpdatePurchaseOrderApprovalRuleInput!): UpdatePurchaseOrderApprovalRuleOutput @doc(description: "Update existing purchase order approval rule")
+    deletePurchaseOrderApprovalRule(input: DeletePurchaseOrderApprovalRuleInput!): DeletePurchaseOrderApprovalRuleOutput @doc(description: "Delete existing purchase order approval rule")
+}
+
+type CreatePurchaseOrderApprovalRuleOutput {
+    approval_rule: PurchaseOrderApprovalRule @doc(description: "Created purchase order approval rule")
+    approval_rules(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderApprovalRules @doc(description: "A list of purchase order approval rules visible to the customer")
+}
+
+type UpdatePurchaseOrderApprovalRuleOutput {
+    approval_rule: PurchaseOrderApprovalRule @doc(description: "Updated purchase order approval rule")
+    approval_rules(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderApprovalRules @doc(description: "A list of purchase order approval rules visible to the customer")
+}
+
+type DeletePurchaseOrderApprovalRuleOutput {
+    approval_rules(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderApprovalRules @doc(description: "A list of purchase order approval rules visible to the customer")
+}
+
+input CreatePurchaseOrderApprovalRuleInput {
+    approval_rule: PurchaseOrderApprovalRuleInput! @doc(description: "Purchase order approval rule data")
+}
+
+input UpdatePurchaseOrderApprovalRuleInput {
+    approval_rule_uid: ID! @doc(description: "Purchase order approval rule ID")
+    approval_rule: PurchaseOrderApprovalRuleInput! @doc(description: "Purchase order approval rule data")
+}
+
+input DeletePurchaseOrderApprovalRuleInput {
+    approval_rule_uid: ID! @doc(description: "Purchase order approval rule ID")
+}
+
+input PurchaseOrderApprovalRuleInput {
+    name: String! @doc(description: "Purchase order approval rule name")
+    description: String @doc(description: "Purchase order approval rule description")
+    applies_to: [ID]! @doc(description: "A list of B2B user roles to which this purchase order approval rule should be applied. In case when empty array is provided, the rule will be applied to all user roles in the system, including those created in the future")
+    type: PurchaseOrderApprovalRuleType! @doc(description: "Purchase order approval rule type")
+    condition: CreatePurchaseOrderApprovalRuleConditionInput! @doc(description: "Purchase order approval rule condition")
+    requires_approval_from: [ID!]! @doc(description: "A list of B2B user roles that can approve this purchase order approval rule")
```

Comment:
With this schema, a client could technically pass in `[]` for `requires_approval_from`, which I believe is a mandatory value on the backend (maybe I'm wrong though, please confirm).

Is it worth leaving an inline comment here with an implementation note advising to throw a validation errors on an empty list?